tests/kernel/mem_protect/obj_validation/ fails to build

commit db094b8d885196215473485093a7f03c90e421c6 breaking tests, for example:

tests/kernel/mem_protect/obj_validation/

```
ccache /home/nashif/Work/SDK/0.9.2/sysroots/x86_64-pokysdk-linux/usr/bin/arm-zephyr-eabi/arm-zephyr-eabi-gcc -DKERNEL -DNRF52832_XXAA -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I../../../../../../../kernel/include -I../../../../../../../arch/arm/include -I../../../../../../../arch/arm/soc/nordic_nrf5/nrf52 -I../../../../../../../arch/arm/soc/nordic_nrf5/nrf52/include -I../../../../../../../arch/arm/soc/nordic_nrf5/include -I../../../../../../../boards/arm/96b_nitrogen -I../../../../../../../include -I../../../../../../../include/drivers -Izephyr/include/generated -isystem /home/nashif/Work/SDK/0.9.2/sysroots/x86_64-pokysdk-linux/usr/lib/arm-zephyr-eabi/gcc/arm-zephyr-eabi/6.2.0/include -isystem /home/nashif/Work/SDK/0.9.2/sysroots/x86_64-pokysdk-linux/usr/lib/arm-zephyr-eabi/gcc/arm-zephyr-eabi/6.2.0/include-fixed -I../../../../../../../lib/libc/minimal/include -I../../../../../../../ext/hal/cmsis/Include -I../../../../../../../ext/hal/nordic/nrfx -I../../../../../../../ext/hal/nordic/nrfx/drivers/include -I../../../../../../../ext/hal/nordic/nrfx/hal -I../../../../../../../ext/hal/nordic/nrfx/mdk -I../../../../../../../ext/hal/nordic/. -I../../../../../../../subsys/bluetooth -I../../../../../../../tests/include -I../../../../../../../tests/ztest/include -Os -g -Wall -Wformat -Wformat-security -Wno-format-zero-length -imacros /home/nashif/Work/github/zephyr/sanity-out/96b_nitrogen/tests/kernel/mem_protect/obj_validation/kernel.memory_protection.obj_validation/zephyr/include/generated/autoconf.h -ffreestanding -Wno-main -std=c99 -Werror -DCONFIG_ASSERT=1 -D__ASSERT_ON=2 -Wno-deprecated-declarations -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -Wno-pointer-sign -Wno-unused-but-set-variable -fno-reorder-functions -fno-defer-pop -Werror=implicit-int -ffunction-sections -fdata-sections -mabi=aapcs --sysroot /home/nashif/Work/SDK/0.9.2/sysroots/armv5-zephyr-eabi/usr -mthumb -mcpu=cortex-m4 -MD -MT zephyr/kernel/CMakeFiles/kernel.dir/mem_domain.c.obj -MF zephyr/kernel/CMakeFiles/kernel.dir/mem_domain.c.obj.d -o zephyr/kernel/CMakeFiles/kernel.dir/mem_domain.c.obj   -c /home/nashif/Work/github/zephyr/kernel/mem_domain.c
In file included from ../../../../../../../include/arch/cpu.h:15:0,
                 from ../../../../../../../include/kernel.h:33,
                 from ../../../../../../../include/device.h:11,
                 from ../../../../../../../include/init.h:11,
                 from /home/nashif/Work/github/zephyr/kernel/mem_domain.c:7:
/home/nashif/Work/github/zephyr/kernel/mem_domain.c: In function ‘sane_partition’:
/home/nashif/Work/github/zephyr/kernel/mem_domain.c:25:43: error: ‘const struct k_mem_partition’ has no member named ‘attrs’; did you mean ‘attr’?
  exec = K_MEM_PARTITION_IS_EXECUTABLE(part->attrs);
                                           ^
../../../../../../../include/arch/arm/arch.h:277:6: note: in definition of macro ‘K_MEM_PARTITION_IS_EXECUTABLE’
  (!((attr) & (NOT_EXEC)))
      ^~~~
/home/nashif/Work/github/zephyr/kernel/mem_domain.c:26:42: error: ‘const struct k_mem_partition’ has no member named ‘attrs’; did you mean ‘attr’?
  write = K_MEM_PARTITION_IS_WRITABLE(part->attrs);
                                          ^
../../../../../../../include/arch/arm/arch.h:265:11: note: in definition of macro ‘K_MEM_PARTITION_IS_WRITABLE’
   switch (attr) { \
           ^~~~
/home/nashif/Work/github/zephyr/kernel/mem_domain.c:44:51: error: ‘const struct k_mem_partition’ has no member named ‘attrs’; did you mean ‘attr’?
   cur_write = K_MEM_PARTITION_IS_WRITABLE(parts[i].attrs);
                                                   ^
../../../../../../../include/arch/arm/arch.h:265:11: note: in definition of macro ‘K_MEM_PARTITION_IS_WRITABLE’
   switch (attr) { \
           ^~~~
/home/nashif/Work/github/zephyr/kernel/mem_domain.c:45:52: error: ‘const struct k_mem_partition’ has no member named ‘attrs’; did you mean ‘attr’?
   cur_exec = K_MEM_PARTITION_IS_EXECUTABLE(parts[i].attrs);
                                                    ^
../../../../../../../include/arch/arm/arch.h:277:6: note: in definition of macro ‘K_MEM_PARTITION_IS_EXECUTABLE’
  (!((attr) & (NOT_EXEC)))
      ^~~~
In file included from ../../../../../../../include/kernel.h:24:0,
                 from ../../../../../../../include/device.h:11,
                 from ../../../../../../../include/init.h:11,

```

--- a/kernel/mem_domain.c
+++ b/kernel/mem_domain.c
@@ -22,8 +22,8 @@ static bool sane_partition(const struct k_mem_partition *part,
 	u32_t i;
 
 	end = part->start + part->size;
-	exec = K_MEM_PARTITION_IS_EXECUTABLE(part->attrs);
-	write = K_MEM_PARTITION_IS_WRITABLE(part->attrs);
+	exec = K_MEM_PARTITION_IS_EXECUTABLE(part->attr);
+	write = K_MEM_PARTITION_IS_WRITABLE(part->attr);
 
 	if (exec && write) {
 		__ASSERT(0, "partition is writable and executable <start %x>",
@@ -41,8 +41,8 @@ static bool sane_partition(const struct k_mem_partition *part,
 			continue;
 		}
 
-		cur_write = K_MEM_PARTITION_IS_WRITABLE(parts[i].attrs);
-		cur_exec = K_MEM_PARTITION_IS_EXECUTABLE(parts[i].attrs);
+		cur_write = K_MEM_PARTITION_IS_WRITABLE(parts[i].attr);
+		cur_exec = K_MEM_PARTITION_IS_EXECUTABLE(parts[i].attr);
 
 		if ((cur_write && exec) || (cur_exec && write)) {
 			__ASSERT(0, "overlapping partitions are "
@@ -96,7 +96,7 @@ void k_mem_domain_init(struct k_mem_domain *domain, u8_t num_parts,
 #if defined(CONFIG_EXECUTE_XOR_WRITE)
 		for (i = 0; i < num_parts; i++) {
 			__ASSERT(sane_partition_domain(domain,
-						       domain->partitions[i]),
+						       &domain->partitions[i]),
 				 "");
 		}
 #endif

